### PR TITLE
Update nfcens to work with ecflow workflow

### PR DESCRIPTION
**Pull request by lgannoaa**
GitHub #196 

Tested in development ecflow workflow. Result reproduced the emc.vpppg parallel.

- [Yes] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

### CODE REVIEW

File: jobs/nfcens/plots/JEVS_NFCENS_WAVE_GRID2OBS_PLOTS [hunk 1]
```diff
@@ -77,12 +75,9 @@ export jlogfile=${DATA}/${MODELNAME}_${RUN}_plots.log
 ####################################
 # Execute the script                
 ####################################
-env
-$HOMEevs/scripts/${COMPONENT}/${STEP}/exevs_${COMPONENT}_${RUN}_${VERIF_CASE}_${STEP}.sh 1>>$pgmout 2>errfile.${pid}
+$HOMEevs/scripts/nfcens/plots/exevs_nfcens_wave_grid2obs_plots.sh
```
Comment (ShelleyMelchior-NOAA) [inline jobs/nfcens/plots/JEVS_NFCENS_WAVE_GRID2OBS_PLOTS]: @lgannoaa , can you update to use $COMPONENT in place of nfcens, $STEP in place of plots? I would also like to see $RUN in place of wave and VERIF_CASE in place of grid2obs.

File: jobs/nfcens/prep/JEVS_NFCENS_WAVE_GRID2OBS_PREP [hunk 2]
```diff
@@ -65,26 +58,23 @@ export VERIF_CASE=${VERIF_CASE:-grid2obs}
...
+mkdir -p ${COMOUT}.${INITDATE}/${MODELNAME}/${VERIF_CASE}
 
 export jlogfile=${DATA}/${MODELNAME}_${RUN}_PB2NC.log
 
 ####################################
 # Execute the script                
 ####################################
-env
-$HOMEevs/scripts/${COMPONENT}/${STEP}/exevs_${COMPONENT}_${RUN}_${VERIF_CASE}_${STEP}.sh 1>>$pgmout 2>errfile.${pid}
+$HOMEevs/scripts/nfcens/prep/exevs_nfcens_wave_grid2obs_prep.sh
```
Comment (ShelleyMelchior-NOAA) [inline jobs/nfcens/prep/JEVS_NFCENS_WAVE_GRID2OBS_PREP]: @lgannoaa , can you update to use $COMPONENT in place of nfcens, $STEP in place of prep? I would also like to see $RUN in place of wave and VERIF_CASE in place of grid2obs.

File: jobs/nfcens/stats/JEVS_NFCENS_WAVE_GRID2OBS_STATS [hunk 3]
```diff
@@ -79,14 +73,9 @@ export jlogfile=${DATA}/${MODELNAME}_${RUN}_PointStat.log
 ####################################
 # Execute the script                
 ####################################
-env
-
-$HOMEevs/scripts/${COMPONENT}/${STEP}/exevs_${COMPONENT}_${RUN}_${VERIF_CASE}_${STEP}.sh 1>>$pgmout 2>errfile.${pid}
-
+$HOMEevs/scripts/nfcens/stats/exevs_nfcens_wave_grid2obs_stats.sh
```
Comment (ShelleyMelchior-NOAA) [inline jobs/nfcens/stats/JEVS_NFCENS_WAVE_GRID2OBS_STATS]: @lgannoaa , can you update to use $COMPONENT in place of nfcens, $STEP in place of stats? I would also like to see $RUN in place of wave and VERIF_CASE in place of grid2obs.

**Comment by lgannoaa** [comment]
@ShelleyMelchior-NOAA Updated and merged upstream.

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @lgannoaa -- can you make another modification for this PR? The ush script, `ush/nfcens/settings.py`, needs a hardcoded development path adjusted. The lines presently look like this:
`self.logo_left_path = "/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix/logos/noaa.png"`
`self.logo_right_path = "/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix/logos/nws.png"`
They need to be adjusted to:
`self.logo_left_path = f"{os.environ['FIXevs']}/logos/noaa.png"`
`self.logo_right_path = f"{os.environ['FIXevs']}/logos/nws.png"`

**Comment by lgannoaa** [comment]
Modification and merge upstream completed.

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks @lgannoaa , one more merge upstream would be great. I am getting started on this PR.

File: jobs/nfcens/prep/JEVS_NFCENS_WAVE_GRID2OBS_PREP [hunk 4]
```diff
@@ -65,26 +58,23 @@ export VERIF_CASE=${VERIF_CASE:-grid2obs}
 # Define COMIN/COMOUT variables     
 ####################################
 export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
-export COMINmodel=${COMINmodel:-$(compath.py ${envir}/com/${RUN}_${MODELNAME}/${model_ver})} 
+export COMINmodel=${COMINmodel:-$(compath.py ${envir}/com/${RUN}_${MODELNAME}/${nfcens_ver})} 
 export COMINgdas=${COMINobs:-$(compath.py ${envir}/com/obsproc/${obsproc_ver}/gdas)}
 
-export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver)}
-export ARCmodel=${COMOUT}/$STEP/$COMPONENT/${RUN}.${INITDATE}/${MODELNAME}/${VERIF_CASE}
+export COMOUT=${COMIN}/${STEP}/${COMPONENT}/${RUN}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/nfcens/prep/JEVS_NFCENS_WAVE_GRID2OBS_PREP]: @lgannoaa , this line is troublesome. The COMOUT variable needs to be able to be overwritten from the driver script. Can this be changed to something like this?
`export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}/${RUN}}`

Comment (LinGan-NOAA) [reply jobs/nfcens/prep/JEVS_NFCENS_WAVE_GRID2OBS_PREP]: Modified as requested.

**Comment by ShelleyMelchior-NOAA** [comment]
OK, back on this PR. Can you do another merge w/ upstream?

**Comment by ShelleyMelchior-NOAA** [comment]
@lgannoaa , I have been running nfcens prep, stats, and plots.

- The stats runs correctly and matches nicely w/ the EVS parallel.
- The prep has a problem in `exevs_nfcens_wave_grid2obs_prep.sh` with copying GDAS ncfiles to the final COMOUT dir. I have corrected it locally and the re-run was successful. The output prep files match nicely w/ the EVS parallel.
- The plots runs, but it produces empty tar files. I found that `import os` needs to be added to the `ush/nfcens/settings.py` script. I have corrected it locally and the re-run was successful. The output tar files match nicely w/ the EVS parallel.

I will update your feature branch w/ the changes I made to the following files:
`scripts/nfcens/prep/exevs_nfcens_wave_grid2obs_prep.sh`
`ush/nfcens/settings.py`

**Comment by ShelleyMelchior-NOAA** [comment]
I think this PR looks good now and I am happy to merge if you see no further problems, @lgannoaa .

**Comment by lgannoaa** [comment]
I am ok

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
Suggested code changes work as expected. This PR is ready for merge.

**Pull request merged**